packages/@aws-cdk/aws-stepfunctions-tasks/lib/eks/call.ts

```diff
@@ -0,0 +1,122 @@
+import * as eks from '@aws-cdk/aws-eks';
+import * as iam from '@aws-cdk/aws-iam';
+import * as sfn from '@aws-cdk/aws-stepfunctions';
+import { Construct } from 'constructs';
+import { integrationResourceArn, validatePatternSupported } from '../private/task-utils';
+
+/**
+ * Properties for calling a EKS endpoint with EksCall
+ * @experimental
+ */
+export interface EksCallProps extends sfn.TaskStateBaseProps {
+
+  /**
+   * The EKS cluster
+   */
+  readonly cluster: eks.ICluster;
+
+  /**
+   * HTTP method ("GET", "POST", "PUT", ...) part of HTTP request
+   */
+  readonly httpMethod: HttpMethods;
+
+  /**
+   * HTTP path of the Kubernetes REST API operation
+   * For example: /api/v1/namespaces/default/pods
+   */
+  readonly httpPath: string;
+
+  /**
+   * Query Parameters part of HTTP request
+   * @default - no query parameters
+   */
+  readonly queryParameters?: { [key: string]: string[] };
+
+  /**
+   * Request body part of HTTP request
+   * @default - No request body
+   */
+  readonly requestBody?: sfn.TaskInput;
+}
+
+/**
+ * Call a EKS endpoint as a Task
+ *
+ * @see https://docs.aws.amazon.com/step-functions/latest/dg/connect-eks.html
+ * @experimental
+ */
+export class EksCall extends sfn.TaskStateBase {
+
+  private static readonly SUPPORTED_INTEGRATION_PATTERNS: sfn.IntegrationPattern[] = [
+    sfn.IntegrationPattern.REQUEST_RESPONSE,
+  ];
+
+  /** No policies are required due to eks:call is an Http service integration and does not call and EKS API directly
+   * @see https://docs.aws.amazon.com/step-functions/latest/dg/connect-eks.html#connect-eks-permissions
+   */
+  protected readonly taskMetrics?: sfn.TaskMetricsConfig;
+  protected readonly taskPolicies?: iam.PolicyStatement[];
+
+  private readonly integrationPattern: sfn.IntegrationPattern;
+
+  constructor(scope: Construct, id: string, private readonly props: EksCallProps) {
+    super(scope, id, props);
+    this.integrationPattern = props.integrationPattern ?? sfn.IntegrationPattern.REQUEST_RESPONSE;
+
+    validatePatternSupported(this.integrationPattern, EksCall.SUPPORTED_INTEGRATION_PATTERNS);
+  }
+
+  /**
+   * Provides the EKS Call service integration task configuration
+   * @internal
+   */
+  protected _renderTask(): any {
+    return {
+      Resource: integrationResourceArn('eks', 'call', this.integrationPattern),
+      Parameters: sfn.FieldUtils.renderObject({
+        ClusterName: this.props.cluster.clusterName,
+        CertificateAuthority: this.props.cluster.clusterCertificateAuthorityData,
+        Endpoint: this.props.cluster.clusterEndpoint,
```

Comment:
Are these parameters required for successfully making the call? Note that if `ICluster` is imported, these properties might not be defined, and will throw an exception. 

I would extract these and do:

```ts
try {
  const ca = this.props.cluster.clusterCertificateAuthorityData;
} catch (err) {
  // if its required, throw an error here with the right context
  // if its not required, just pass
}
```

Reply:
Thanks for the feedback, I'll add validation for ClusterName, ClusterEndpoint and clusterCertificateAuthorityData, as well as coverage tests for these cases

Reply:
With some testing I think the error thrown from the eks module has the valid context in our case.

for eks:call I'm extracting ```clusterName```, ```clusterEndpoint```, and ```clusterCertificateAuthorityData``` from ICluster, for example if clusterEndpoint is not defined, it will throw exception with message ```"clusterEndpoint" is not defined for this imported cluster```, which is sufficient to show the context of the error.

All three are required to make the call, I'll just add coverage test to reflect this case

Reply:
@shivlaks Do you think ```"clusterEndpoint" is not defined for this imported cluster``` is sufficient or more details needs to be added such as ```Cluster Endpoint is not defined for this imported cluster, this field is required for eks:call task.```

Reply:
@novakgu that seems sensible. let's keep it simple. I generally like the format of `<validation error><remediation (if any)>`

i.e.
The `clusterEndpoint` property must be specified when using an imported Cluster.

Reply:
Added validation for ```clusterEndpoint``` and ```clusterCertificateAuthorityData```. 
```clusterName``` is required when import a cluster or creating a new cluster so I'm not double checking it here.

After changes 
* if an given cluster does not have cluster Endpoint it will throw with error 
```The "clusterEndpoint" property must be specified when using an imported Cluster.```
* If an given cluster does not have certAuth it will throw with error
```The "clusterCertificateAuthorityData" property must be specified when using an imported Cluster.```